torch/csrc/jit/pybind_utils.h

```diff
@@ -538,6 +538,9 @@ inline IValue toIValue(
         return py::cast<double>(obj);
       }
     }
+    case TypeKind::PyObjectType:
+      // convert a pybind11 object to the IValue and hold it
+      return c10::ivalue::PyObjectHolder::create(obj.ptr());
```

Comment:
There is not incref here. If the python object goes dead then the reference in the by object holder is invalidated.